src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -138,6 +138,13 @@ def __init__(  # pylint: disable=too-many-arguments
         self._resampler_sender: Sender[Sample] | None = None
         self._resampler_task: asyncio.Task[None] | None = None
 
+        self.count_samples = 0
+        """The number of samples that have been received."""
```

Comment:
I think this one should probably be a read-only public attribute. Also for attributes it's better to use nouns instead of verbs (and it's missing the type), for example:

```suggestion
        self._received_samples_count: int = 0
        """The number of samples that have been received."""
```

Plus

```py        
    @property
    def received_samples_count(self) -> int
        return self._received_samples_count
```